File: backend/src/models/Adjunto.js
const { pool } = require('../config/db');

class Adjunto {
  /**
   * Registrar un nuevo archivo adjunto asociado a un trámite o movimiento
   */
  static async create({
    tramite_id,
    movimiento_id = null,
    nombre_original,
    nombre_almacenado,
    ruta_archivo,
    tipo_mime = 'application/pdf',
    tamano_bytes = 0,
    subido_por
  }) {
    const [result] = await pool.query(`
      INSERT INTO adjuntos (
        tramite_id, movimiento_id, nombre_original, nombre_almacenado,
        ruta_archivo, tipo_mime, tamano_bytes, subido_por, activo
      )
      VALUES (?, ?, ?, ?, ?, ?, ?, ?, 1)
    `, [
      tramite_id,
      movimiento_id,
      nombre_original,
      nombre_almacenado,
      ruta_archivo,
      tipo_mime,
      tamano_bytes,
      subido_por
    ]);

    return this.findById(result.insertId);
  }

  /**
   * Buscar adjunto por ID con datos del usuario que lo subió
   */
  static async findById(id) {
    const [rows] = await pool.query(`
      SELECT 
        a.*,
        u.login as subido_por_login,
        p.nombres, p.apellido_paterno
      FROM adjuntos a
      INNER JOIN usuarios u ON a.subido_por = u.id
      INNER JOIN personas p ON u.persona_id = p.id
      WHERE a.id = ?
    `, [id]);

    return rows[0] || null;
  }

  /**
   * Listar todos los adjuntos de un trámite
   */
  static async findByTramiteId(tramite_id, { activo = true } = {}) {
    let sql = `
      SELECT 
        a.*,
        u.login as subido_por_login,
        p.nombres, p.apellido_paterno
      FROM adjuntos a
      INNER JOIN usuarios u ON a.subido_por = u.id
      INNER JOIN personas p ON u.persona_id = p.id
      WHERE a.tramite_id = ?
    `;
    const params = [tramite_id];

    if (activo !== null) {
      sql += ' AND a.activo = ?';
      params.push(activo ? 1 : 0);
    }

    sql += ' ORDER BY a.id ASC';
    const [rows] = await pool.query(sql, params);
    return rows;
  }

  /**
   * Listar adjuntos de un movimiento específico
   */
  static async findByMovimientoId(movimiento_id, { activo = true } = {}) {
    let sql = `
      SELECT 
        a.*,
        u.login as subido_por_login,
        p.nombres, p.apellido_paterno
      FROM adjuntos a
      INNER JOIN usuarios u ON a.subido_por = u.id
      INNER JOIN personas p ON u.persona_id = p.id
      WHERE a.movimiento_id = ?
    `;
    const params = [movimiento_id];

    if (activo !== null) {
      sql += ' AND a.activo = ?';
      params.push(activo ? 1 : 0);
    }

    sql += ' ORDER BY a.id ASC';
    const [rows] = await pool.query(sql, params);
    return rows;
  }

  /**
   * Borrado lógico de adjunto
   */
  static async softDelete(id) {
    await pool.query('UPDATE adjuntos SET activo = 0 WHERE id = ?', [id]);
    return true;
  }
}

module.exports = Adjunto;

File: backend-dotnet/GestionDocumental.Api/Entities/Adjunto.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace GestionDocumental.Api.Entities
{
    [Table("adjuntos")]
    public class Adjunto
    {
        [Key]
        [Column("id")]
        public int Id { get; set; }

        [Column("tramite_id")]
        public int TramiteId { get; set; }

        [Column("movimiento_id")]
        public int? MovimientoId { get; set; }

        [Required]
        [MaxLength(255)]
        [Column("nombre_original")]
        public string NombreOriginal { get; set; } = string.Empty;

        [Required]
        [MaxLength(255)]
        [Column("nombre_almacenado")]
        public string NombreAlmacenado { get; set; } = string.Empty;

        [Required]
        [MaxLength(100)]
        [Column("mime_type")]
        public string MimeType { get; set; } = "application/pdf";

        [Column("tamano_bytes")]
        public long TamanoBytes { get; set; }

        [Required]
        [MaxLength(500)]
        [Column("ruta_archivo")]
        public string RutaArchivo { get; set; } = string.Empty;

        [MaxLength(64)]
        [Column("hash_sha256")]
        public string? HashSha256 { get; set; }

        [Column("subido_por")]
        public int SubidoPor { get; set; }

        [Column("activo")]
        public bool Activo { get; set; } = true;

        [Column("created_at")]
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        // Relaciones
        [ForeignKey("TramiteId")]
        public virtual Tramite Tramite { get; set; } = null!;

        [ForeignKey("MovimientoId")]
        public virtual Movimiento? Movimiento { get; set; }

        [ForeignKey("SubidoPor")]
        public virtual Usuario SubidoPorUsuario { get; set; } = null!;
    }
}
